### ISSUE TITLE
Ordinal Architecture for Discretizing Continuous Action Spaces

### ISSUE THREAD

**Issue by ruanrothmann** [Unity-Technologies/ml-agents#1733]
Hi, this isn't really an issue, so apologies if this is not the right place for this kind of question.

This paper discusses how one might discretize a continuous action space and how that might improve training results for things like humanoid walker etc: https://robintyh1.github.io/papers/tang2018discretizing.pdf#page=4

I understand the theory & practice behind what they are saying and it looks relatively straightforward to retool an existing continuous agent as discrete. Unfortunately, I don't have the academic background (or im just too dumb lol)  to understand section 4 at all (referring to the "ordinal architecture").

Is what the writer is describing there analogous to simply using an action branch for each continuous action dimension? Or is he referring to some sort of modification of the network? Is this something I can attempt in UnityML? Besides just making a discrete action size of ~7^36 in my case.

**Comment by vincentpierre** [comment]
Hi, I have not read the paper but I think it is implementing multi-discrete in a different way than we do. You could try to implement it and make it a contribution or you could try to use our implementation of multi-discrete. 
If you want to use our multi-discrete then yes, each continuous action would translate to one branch in the discrete scenario. How much you want to discretize is up to you but I would say ~10 or 20.

**Comment by ruanrothmann** [comment]
What I'm curious about: Is there any difference in how training is performed with action branches vs just having one branch?

For instance, if I have 2 branches each with size 10, is there any real difference in training that vs a training setup with 1 branch of size 100?

I'm asking because, the total action space for a discretized setup gets massive very quick - 7^36 to be exact. If I were to perform training for 100 million steps, that's still WAY fewer steps than the size of the total action space.  

From my reading I understand that at least OpenAI 5 use this sort of discretized action space for e.g. issuing move commands. But I imagine in their setup there is a lot of masking, which probably cuts down on the action space per step quite significantly.

**Comment by vincentpierre** [comment]
Thank you for the discussion. We are closing this issue due to inactivity.
